Backpack/Snackbar/Classes/BPKSnackbar.m

```diff
@@ -0,0 +1,370 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "BPKSnackbar.h"
+
+#import <Backpack/Button.h>
+#import <Backpack/Color.h>
+#import <Backpack/Label.h>
+#import <Backpack/Spacing.h>
+
+NSString *const SnackbarAccessibilityIdentifier = @"snackbarView";
+
+@interface BPKSnackbar ()
+@property(strong, nonatomic) BPKLabel *titleLabel;
+@property(strong, nonatomic) BPKButton *actionButton;
+@property(strong, nonatomic) UIImageView *leftIconContainer;
+@property(strong, nonatomic) UIView *snackbarView;
+@property(weak, nonatomic) id<BPKSnackbarProtocol> delegate;
+@property(nonatomic) BPKSnackbarDuration duration;
+@property(nonatomic, strong) NSTimer *timer;
+@property(nonatomic) long durationInMillis;
+@property(nonatomic) CGFloat keyboardHeight;
+@property(nonatomic, strong) NSLayoutConstraint *bottomConstraint;
+@property(nonatomic, strong) NSLayoutConstraint *heightConstraint;
+@property(nonatomic, strong) NSLayoutConstraint *snackbarViewTopConstraint;
+@property(nonatomic, strong) NSLayoutConstraint *labelButtonSpacing;
+@property(nonatomic, strong) NSLayoutConstraint *labelLeadingSpacing;
+@property(nonatomic, strong) UIViewController *parentViewController;
+
+@end
+
+static CGFloat const BPKSnackbarAnimationDuration = 0.3;
```

Comment:
We have tokens for durations in Backpack
https://backpack.github.io/ios/versions/23.2.0/Classes/BPKDuration.html#/c:objc(cs)BPKDuration(cpy)animationDurationBase